File: rest-service/src/main/java/mb/lib/formatter/db/SelectAllJobReports.java
package mb.lib.formatter.db;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.util.List;

import io.vulpine.lib.query.util.basic.BasicPreparedListReadQuery;
import mb.lib.formatter.model.ReportRow;
import mb.lib.model.HashID;

public class SelectAllJobReports
{
  private static final String Query = """
    SELECT
      *
    FROM
      userlogins5.multiblast_fmt_jobs
    WHERE
      job_digest = ?
    """;

  private final Connection con;
  private final HashID jobID;

  public SelectAllJobReports(Connection con, HashID jobID) {
    this.con   = con;
    this.jobID = jobID;
  }

  public List<ReportRow> run() throws Exception {
    return new BasicPreparedListReadQuery<>(Query, con, Util::rs2Row, this::prep)
      .execute()
      .getValue();
  }

  private void prep(PreparedStatement ps) throws Exception {
    ps.setBytes(1, jobID.bytes());
  }
}

File: rest-service/src/main/java/mb/lib/report/db/SelectAllJobReports.kt
package mb.lib.report.db

import io.vulpine.lib.query.util.basic.BasicPreparedListReadQuery
import mb.lib.model.HashID
import mb.lib.report.model.ReportRow
import java.sql.Connection
import java.sql.PreparedStatement

data class SelectAllJobReports(val con: Connection, val jobID: HashID) {
  companion object {
    private const val Query = """
    SELECT
      *
    FROM
      userlogins5.multiblast_fmt_jobs
    WHERE
      job_digest = ?
    """
  }
  fun run(): List<ReportRow> = BasicPreparedListReadQuery(Query, con, ::rs2Row, this::prep).execute().value!!

  fun prep(ps: PreparedStatement) = ps.setBytes(1, jobID.bytes)
}
